### ISSUE TITLE
Revisions don't seem to be working in 1.8.9

### ISSUE THREAD

**Issue by MaratBR** [roman-right/beanie#179]
Sorry for slightly vague title. Consider this code:

```python3
import asyncio

import motor as motor
from beanie import init_beanie, Document
from beanie.exceptions import RevisionIdWasChanged


class Sample(Document):
    num: int
    name: str

    class Settings:
        use_revision = True


async def do():
    s = await Sample.find_one({'num': 42})

    if s is None:
        print('creating Sample')
        s = Sample(num=42, name='Hi')
        await s.save()
    else:
        print(f'Modifying Sample, before replace(): revision_id={s.revision_id} _previous_revision_id={s._previous_revision_id}')
        s.name += '1'
        await s.replace()
        print(f'After replace(): revision_id={s.revision_id} _previous_revision_id={s._previous_revision_id}')


async def issue():
    # Crete Motor client
    client = motor.motor_asyncio.AsyncIOMotorClient(
        "mongodb://USERNAME:PASSWORD@HOST:PORT"
    )

    # Init beanie with the Product document class
    await init_beanie(database=client.test, document_models=[Sample])

    await Sample.delete_all()

    print('calling do() once')
    await do()  # ok
    print('calling do() twice')
    await do()  # ok, after second call revision_id is set in database (i.e. not null)
    print('calling do() thrice, here it comes')
    try:
        await do()  # RevisionIdWasChanged
    except RevisionIdWasChanged:
        print('RevisionIdWasChanged')

asyncio.get_event_loop().run_until_complete(issue())
```
Output:
```
calling do() once
creating Sample
calling do() twice
Modifying Sample, before replace(): revision_id=c3eb91c4-6c2c-4931-b0d0-0286fc6dcfc0 _previous_revision_id=None
After replace(): revision_id=2c2246f8-00fb-4c24-9954-86b54e9032c4 _previous_revision_id=c3eb91c4-6c2c-4931-b0d0-0286fc6dcfc0
calling do() thrice, here it comes
Modifying Sample, before replace(): revision_id=c3eb91c4-6c2c-4931-b0d0-0286fc6dcfc0 _previous_revision_id=None
RevisionIdWasChanged
```

**Expected behavior:**
Sample is created and then modified 2 times sequentially, and, therefore, no race condition.

**Actual behavior:**
Sample is created, then modified once and then on the third call of `do()` (second modification) `RevisionIdWasChanged` is raised.

**Some info:**
pip freeze:
```
beanie==1.8.9
click==7.1.2
idna==3.3
motor==2.5.1
multidict==5.2.0
pydantic==1.8.2
pymongo==3.12.3
toml==0.10.2
typing_extensions==4.0.1
yarl==1.7.2
```
Python version: 3.8.10
OS: Linux

**Comment by MaratBR** [comment]
Oh, and this bug was introduced in 1.8.9, downgrading to 1.8.8. fixes it

**Comment by roman-right** [comment]
Hey @MaratBR ,
This is critical. Thank you.
Will be fixed today.

**Comment by roman-right** [comment]
Should be fixed in 1.8.10
Pls, check.

**Comment by MaratBR** [comment]
Yes, it's fixed now. Thank you for fast reply!
